RetailCoder.VBE/Refactorings/Rename/RenameModel.cs

```diff
@@ -76,5 +108,41 @@ public void PromptIfTargetImplementsInterface(ref Declaration target)
 
             target = interfaceMember;
         }
+
+        internal class Rewriters
```

Comment:
Thinking about this, I believe it is necessary for Rename to cache the QMNs (at a minimum) because as the Rename process goes through all the references, implementations, and declaration, it could modify 1 to n modules.  And the same module may be updated more than once.  After all the relevant TSRs have been modified, the Rewrite would operate on all the edited modules in the 'final' operation.  Waiting to rewrite also seems appropriate because if I rewrote as I go, an exception between rewrites could leave the module in a bad state.  Even with that in mind, I can see now that I called Rewrite more frequently than needed.  I was retaining the 'rule' in the original code of writing declarations last...but it seems with TSR I can do all rewrites at once - in any order.